cfn-resources/stream-instance/cmd/resource/resource.go

```diff
@@ -0,0 +1,318 @@
+	setup()
+	util.SetDefaultProfileIfNotDefined(&currentModel.Profile)
+	if errEvent := validator.ValidateModel(DeleteRequiredFields, currentModel); errEvent != nil {
+		return *errEvent, nil
+	}
+
+	client, handlerError := util.NewAtlasV2OnlyClientLatest(&req, currentModel.Profile, true)
+	if handlerError != nil {
+		_, _ = logger.Warnf("CreateMongoDBClient error: %v", handlerError)
+		return *handlerError, errors.New(handlerError.Message)
+	}
+
+	atlasV2 := client.AtlasSDK
+
+	_, resp, err := atlasV2.StreamsApi.DeleteStreamInstance(context.Background(), *currentModel.GroupId, *currentModel.Name).Execute()
+	if err != nil {
+		return handleError(resp, constants.DELETE, err)
+	}
+
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Delete success"}, nil
+}
+
+func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+	util.SetDefaultProfileIfNotDefined(&currentModel.Profile)
+	if errEvent := validator.ValidateModel(ListRequiredFields, currentModel); errEvent != nil {
+		return *errEvent, nil
+	}
+
+	client, handlerError := util.NewAtlasV2OnlyClientLatest(&req, currentModel.Profile, true)
+	if handlerError != nil {
+		_, _ = logger.Warnf("CreateMongoDBClient error: %v", handlerError)
+		return *handlerError, errors.New(handlerError.Message)
+	}
+
+	atlasV2 := client.AtlasSDK
+
+	accumulatedStreamInstances := make([]admin.StreamsTenant, 0)
+	pageNum := 1
+	for ok := true; ok; {
+		listStreamInstancesRequest := atlasV2.StreamsApi.ListStreamInstances(context.Background(), *currentModel.GroupId)
+		listStreamInstancesRequest.PageNum(pageNum)
+		listStreamInstancesRequest.ItemsPerPage(100)
+		streamInstances, resp, err := listStreamInstancesRequest.Execute()
+		if err != nil {
+			return handleError(resp, constants.LIST, err)
+		}
+		accumulatedStreamInstances = append(accumulatedStreamInstances, streamInstances.GetResults()...)
+		ok = streamInstances.GetTotalCount() > len(accumulatedStreamInstances)
+		pageNum++
+	}
+
+	response := make([]interface{}, 0)
+	for _, stream := range accumulatedStreamInstances {
+		cloudProvider := stream.DataProcessRegion.CloudProvider
+		region := stream.DataProcessRegion.Region
+		model := Model{
+			Name: stream.Name,
+			DataProcessRegion: &StreamsDataProcessRegion{
+				CloudProvider: &cloudProvider,
+				Region:        &region,
+			},
+			StreamConfig: &StreamConfig{
+				Tier: stream.StreamConfig.Tier,
+			},
+			GroupId:     stream.GroupId,
+			Id:          stream.Id,
+			Hostnames:   *stream.Hostnames,
+			Profile:     currentModel.Profile,
+			Connections: newModelConnections(stream.Connections),
+		}
+		response = append(response, model)
+	}
+
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "List Complete",
+		ResourceModels:  response,
+	}, nil
+}
+
+func newStreamsTenant(model *Model) *admin.StreamsTenant {
```

Comment:
do we plan to add unit tests as part of this PR? Additionally we could think of defining conversions method in a separate file similar to our terraform repository. Cluster resource has a https://github.com/mongodb/mongodbatlas-cloudformation-resources/blob/master/cfn-resources/cluster/cmd/resource/mappings.go file for this.

Reply:
added relevant unit tests to some mapping methods. Also moved mapping methods to new file